R/splatPop-simulate.R

```diff
@@ -199,6 +223,61 @@ splatPopSimulateMeans <- function(vcf = mockVCF(),
 }
 
 
+#' splatPopParseEmpirical
+#'
+#' Parse splatPop key information from empirical data provided.
+#' 
+#' NOTE: This function will cause some of the parameters in the splatPopParams 
+#' object to be ignored, such as population level gene mean and variance and 
+#' eQTL parameters. 
+#'
+#' @param vcf VariantAnnotation object containing genotypes of samples.
+#' @param gff Either NULL or a data.frame object containing a GFF/GTF file.
+#' @param eqtl Either NULL or if simulating population parameters directly from
+#'        empirical data, a data.frame with empirical/desired eQTL results.
+#'        To see required format, run `mockEmpiricalSet()` and see eqtl output.
+#' @param means Either NULL or if simulating population parameters directly from
+#'        empirical data, a Matrix of real gene means across a population, where
+#'        each row is a gene and each column is an individual in the population.
+#'        To see required format, run `mockEmpiricalSet()` and see means output.
+#' @param params SplatPopParams object containing parameters for population
+#'        scale simulations. See \code{\link{SplatPopParams}} for details.
+#'
+#' @details This function will ignore a number of parameters defined in 
+#' splatPopParams, instead pulling key information directly from provided VCF,
+#' GFF, gene means, and eQTL mapping result data provided.
+#'
+#' }
+#'
+#' @return A partial splatPop `key` 
+#'
+#' @export
+splatPopParseEmpirical <- function(vcf = vcf, gff = gff, eqtl = eqtl, 
+                                   means = means, params = params){
+    
+    key <- data.frame(list(geneID = eqtl$geneID,
+                           chromosome = gff$V1,
+                           geneStart = gff$V4,
+                           geneEnd = gff$V5,
+                           geneMiddle = floor(abs((gff$V4 - gff$V5)/2)) +
+                               gff$V4,
+                           meanSampled.noOutliers = rowMeans(means),
+                           OutlierFactor = 1,
+                           meanSampled = rowMeans(means),
+                           cvSampled = apply(means, 1, co.var),
+                           eQTL.group = ifelse(is.na(eqtl$snpID), NA, "global"),
+                           eQTL.condition = "global",
+                           eSNP.ID = eqtl$snpID,
+                           eSNP.chromosome = eqtl$snpCHR, 
+                           eSNP.loc = eqtl$snpLOC,
+                           eSNP.MAF = eqtl$snpMAF, 
+                           eQTL.EffectSize = eqtl$slope))
+    
+    params <- setParams(params, nGenes = nrow(key))
```

Comment:
I don't think this is doing anything because `params` isn't returned.

Reply:
Moved it up to the main function so it gets changed (it doesn't have any downstream impact other than making sure the params object reflects the correct number of genes simulated if empirical data is provided).